nixos/modules/services/audio/mympd.nix

```diff
@@ -0,0 +1,83 @@
+{ pkgs, config, lib, ... }:
+
+let
+  cfg = config.services.mympd;
+in {
+  options = {
+
+    services.mympd = {
+
+      enable = lib.mkEnableOption (lib.mdDoc "MyMPD server");
+
+      package = lib.mkPackageOption pkgs "mympd" {};
+
+      port = lib.mkOption {
+        type = lib.types.port;
+        description = lib.mdDoc ''
+          The HTTP port where mympd's web interface will be available.
+
+          The HTTPS/SSL port can be configured via {option}`config`.
+        '';
+        example = "8080";
+      };
+
+      ssl = lib.mkOption {
+        type = lib.types.bool;
+        description = lib.mdDoc ''
+          Whether to enable listening on the SSL port.
+
+          Refer to <https://jcorporation.github.io/myMPD/configuration/configuration-files#ssl-options>
+          for more information.
+        '';
+        default = false;
+      };
+
+      config = lib.mkOption {
+        type = with lib.types; attrsOf (nullOr (oneOf [ str bool int ]));
+        description = lib.mdDoc ''
+          Manages the configuration files declaratively. For all the configuration
+          options, see <https://jcorporation.github.io/myMPD/configuration/configuration-files>.
+
+          Each key represents the "File" column from the upstream configuration table, and the
+          value is the content of that file.
+
+          The configuration is automatically written at service start-up, so it is
+          possible to change their values after the first start of the service.
```

Comment:
It sounds to me a bit as if you suggest to change the values imperatively? I'm not sure what would be the consequences of that, but for sure the values will be overridden when the service restarts, so why imply that?

Reply:
Yes, maybe I was not very clear, I'll try to rephrase it.

The idea is that the upstream has a weird behaviour regarding the treatment of the initial configuration when you use environment variables:

> At the first start (if there is no config folder in the working folder) myMPD reads some environment variables and writes the configuration files.
> To change these settings afterwards, you must edit the files in the folder /var/lib/mympd/config/ and restart myMPD.

What I wanted to highlight is that the NixOS service does not care about that, it will always modify your configuration in `/var/lib/mympd/config`.